routes/api/v1/favorites.js

```diff
@@ -0,0 +1,85 @@
+var express = require('express');
+var router = express.Router();
+
+const environment = process.env.NODE_ENV || 'development';
+const configuration = require('../../../knexfile')[environment];
+const database = require('knex')(configuration);
+const fetch = require('node-fetch');
+
+
+router.post('/', (request, response) => {
+  // 1. check request body comes with location and api_key
+  const info = request.body;
+
+  for (let requiredParameter of ['location', 'api_key']) {
+    if (!info[requiredParameter]) {
+      return response
+        .status(422)
+        .send({ error: `Expected format: { location: <String>, api_key: <String> }. You're missing a "${requiredParameter}" property.` });
+    }
+  }
+  // 2. check api_key in users and return user
+  database('users')
+    .where('api_key', info['api_key'])
+    .select()
+    .first()
+    .then(user => {
+      if(user) {
+        return user
+      } else {
+        return response
+          .status(401)
+          .json({"message": "Unauthorized request"})
+      }
+    })
+    .then(user => {
+      // 3. add favorite with user reference
+      database('favorites').insert({location: info['location'],
+                                    user_id: user.id}, 'id')
+      .then(async(favorite) => {
+        addLatLong(favorite, info['location'])
+          .then(data => {
+            if(data === undefined){
+              // no latLong found
+              console.log(favorite[0], info['location'])
+              database('favorites')
+                .where({ id: favorite[0] })
+                .del()
+                .then(()=>{
+                  return response.status(404).send("Location not found on Google Maps, please enter a different location")
+                })
```

Comment:
This is the extra database call that could use refactoring.

Reply:
For adding favorites maybe sending out an API call isn't necessary and when you go the `get` you could have that as a sad path for one of your multiple objects you're sending back

Reply:
True, that does make sense. But I think I prefer having the google api call in the `post` so that I'm making one api call per location in the `get` instead of 2.